src/racer/core.rs

```diff
@@ -955,6 +956,25 @@ fn complete_from_file_(
             let is_global = expr.starts_with("::");
             let is_use = line.starts_with("use ");
 
+            // when in the function ident position, only look for methods
+            // from a trait to complete.
+            if util::in_fn_name(line) {
+                trace!("Path is in fn declaration: `{}`", expr);
+
+                for m in nameres::resolve_method(
```

Comment:
As part of #778 I changed the return type of this function to be `Vec`, which will remove the `into_iter` thing. There's honestly no need to append in this case, so I can remove the loop outright and just return the `resolve_method` return value directly.